src/server.rs

```diff
@@ -923,14 +917,21 @@ impl LanguageServer for LspServer {
     }
 }
 
+fn file_not_opened(key: &lsp::Url) -> lspower::jsonrpc::Error {
```

Comment:
I'm usually not a fan of abstracting this kind of functionality too much (YAGNI before DRY and all that), but I'm probably good with this one.